libcxx/test/std/algorithms/alg.sorting/alg.sort/stable.sort/stable_sort.pass.cpp

```diff
@@ -80,66 +77,149 @@ test_sort_()
     }
 }
 
-void
-test_larger_sorts(int N, int M)
-{
-    assert(N != 0);
-    assert(M != 0);
-    // create array length N filled with M different numbers
-    int* array = new int[N];
-    int x = 0;
-    for (int i = 0; i < N; ++i)
-    {
-        array[i] = x;
-        if (++x == M)
-            x = 0;
+template <int N, int M>
```

Comment:
Please don't move the individual test into different functions. If you really want to you can do
```c++
{ // description
  <test code>
}
```
but I'd just keep it as-is if there isn't a reason to change things here.

Reply:
This hasn't been addressed.

Reply:
I reverted as much as possible. Some changes were necessary to enable compile-time checks (e.g. passing array sizes as template arguments instead of function arguments).